extension/src/XliffFunctions.ts

```diff
@@ -299,68 +295,62 @@ export function refreshSelectedXlfFileFromGXlf(
           }
           refreshResult.numberOfUpdatedNotes++;
         }
-        if (langTransUnit.sourceIsEmpty() && langTransUnit.targetIsEmpty()) {
+        if (
+          langTransUnit.sourceIsEmpty() &&
+          langTransUnit.targetIsEmpty() &&
+          lfSettings.preferLockedTranslations
+        ) {
           langTransUnit.insertCustomNote(
             CustomNoteType.refreshXlfHint,
             RefreshXlfHint.emptySource
           );
-          refreshResult.numberOfReviewsAdded++;
+          langTransUnit.target.translationToken = TranslationToken.review;
         }
         formatTransUnitForTranslationMode(
-          languageFunctionsSettings.translationMode,
+          lfSettings.translationMode,
           langTransUnit
         );
-        detectInvalidValues(langTransUnit, languageFunctionsSettings);
+        detectInvalidValues(langTransUnit, lfSettings);
         if (langTransUnit.needsReview(true)) {
           refreshResult.numberOfReviewsAdded++;
         }
       }
       newLangXliff.transunit.push(langTransUnit);
       langXliff.transunit.splice(langXliff.transunit.indexOf(langTransUnit), 1); // Remove all handled TransUnits -> The rest will be deleted.
-    } else {
+    } else if (!sortOnly) {
       // Does not exist in target
-      if (!sortOnly) {
-        const newTransUnit = TransUnit.fromString(gTransUnit.toString());
-        newTransUnit.targets = [];
-        newTransUnit.targets.push(
-          getNewTarget(
-            languageFunctionsSettings.translationMode,
-            langIsSameAsGXlf,
-            gTransUnit
-          )
-        );
-        if (langIsSameAsGXlf) {
-          newTransUnit.insertCustomNote(
-            CustomNoteType.refreshXlfHint,
-            RefreshXlfHint.newCopiedSource
-          );
-        } else {
-          newTransUnit.insertCustomNote(
-            CustomNoteType.refreshXlfHint,
-            RefreshXlfHint.new
-          );
-        }
-        if (newTransUnit.sourceIsEmpty()) {
-          newTransUnit.insertCustomNote(
-            CustomNoteType.refreshXlfHint,
-            RefreshXlfHint.emptySource
-          );
-        }
-        formatTransUnitForTranslationMode(
-          languageFunctionsSettings.translationMode,
-          newTransUnit
+      const newTransUnit = TransUnit.fromString(gTransUnit.toString());
+      newTransUnit.targets = [];
+      newTransUnit.targets.push(
+        getNewTarget(lfSettings.translationMode, langIsSameAsGXlf, gTransUnit)
+      );
+      const hintText = langIsSameAsGXlf
+        ? RefreshXlfHint.newCopiedSource
+        : RefreshXlfHint.new;
+      newTransUnit.insertCustomNote(CustomNoteType.refreshXlfHint, hintText);
+      newTransUnit.target.translationToken = TranslationToken.notTranslated;
+
+      if (newTransUnit.sourceIsEmpty() && lfSettings.preferLockedTranslations) {
+        newTransUnit.insertCustomNote(
+          CustomNoteType.refreshXlfHint,
+          RefreshXlfHint.emptySource
```

Comment:
We might consider to add something to the `RefreshXlfHint.emptySource` text that explains how to disable this through the new setting?
Otherwise I think users will freak out, since this note and REVIEW text will come back over and over again :)

Reply:
added